tests/test_s3.py

```diff
@@ -164,6 +176,33 @@ def _check_presigned_url(self, url: str, expiration_time_seconds: int):
         # out to lunch for a while on a busy CI machine
         self.assertLessEqual(abs(expected_expiration_timestamp - actual_expiration_timestamp), 120)
 
+    def test_threading_true(self):
+        os.environ["LSST_S3_USE_THREADS"] = "True"
+        S3ResourcePath.use_threads = None
+        test_resource_path = self.root_uri.join("test_file.dat")
+        self.assertTrue(test_resource_path._transfer_config.use_threads)
+
+    def test_implicit_default_threading(self):
+        S3ResourcePath.use_threads = None
+        boto_default = signature(boto3.s3.transfer.TransferConfig).parameters["use_threads"].default
+        test_resource_path = self.root_uri.join("test_file.dat")
+        self.assertEqual(test_resource_path._transfer_config.use_threads, boto_default)
+
+    def test_explicit_default_threading(self):
+        os.environ["LSST_S3_USE_THREADS"] = "None"
+        S3ResourcePath.use_threads = None
+        boto_default = signature(boto3.s3.transfer.TransferConfig).parameters["use_threads"].default
+        test_resource_path = self.root_uri.join("test_file.dat")
+        self.assertEqual(test_resource_path._transfer_config.use_threads, boto_default)
+
+    def test_threading_false(self):
+        os.environ["LSST_S3_USE_THREADS"] = "False"
```

Comment:
The tests make it clear that we aren't caching the parsing of the environment variable because otherwise this test would fail because it would be cached from a previous read. `@cached_property` is per-instance, so we are reparsing every time. I'm a bit sensitive to this at the moment because I'm looking at overheads for transferring 800k files and things that don't take long add up when there are that many ResourcePath objects.

PS The normal thing in tests is to use:

```python
with mock.patch.dict(os.environ, {"LSST_S3_USE_THREADS": "False"}):
    ...
```

Reply:
Or I can pull the parsing code into a new function (outside of the class), say _parse_use_threading, use a plain cached decorator on it.  The code would then still check the environ var for each instance (but still not each use of each instance), but the parsing code would only happen once (per value the environ takes during the execution, probably only one except for the tests).

Reply:
Ok. Try that.